nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/AzureLogAnalyticsReportingTask.java

```diff
@@ -0,0 +1,335 @@
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.status.ConnectionStatus;
+import org.apache.nifi.controller.status.ProcessGroupStatus;
+import org.apache.nifi.controller.status.ProcessorStatus;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.AbstractReportingTask;
+import org.apache.nifi.reporting.ReportingContext;
+import org.apache.nifi.reporting.azure.loganalytics.api.AzureLogAnalyticsMetricsFactory;
+import org.apache.nifi.scheduling.SchedulingStrategy;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.metrics.jvm.JmxJvmMetrics;
+import org.apache.nifi.metrics.jvm.JvmMetrics;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * ReportingTask to send metrics from Apache NiFi and JVM to Azure Monitor.
+ */
+@Tags({"reporting", "loganalytics", "metrics"})
+@CapabilityDescription("Sends JVM-metrics as well as Apache NiFi-metrics to a Azure Log Analytics workspace." +
+        "Apache NiFi-metrics can be either configured global or on process-group level.")
+@DefaultSchedule(strategy = SchedulingStrategy.TIMER_DRIVEN, period = "1 min")
+public class AzureLogAnalyticsReportingTask extends AbstractReportingTask {
+
+    private static final String JVM_JOB_NAME = "jvm_global";
+    private static final Charset            UTF8                = Charset.forName("UTF-8");
+    private static final String             HMAC_SHA256_ALG     = "HmacSHA256";
+    static final DateTimeFormatter   RFC_1123_DATE_TIME  = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss O");
+    private volatile JvmMetrics virtualMachineMetrics;
+
+
+    static final PropertyDescriptor LOG_ANALYTICS_WORKSPACE_ID = new PropertyDescriptor.Builder()
+            .name("Log Analytics Workspace Id")
+            .description("Log Analytics Workspace Id")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .sensitive(true)
+            .build();
+    static final PropertyDescriptor LOG_ANALYTICS_CUSTOM_LOG_NAME = new PropertyDescriptor.Builder()
+            .name("Log Analytics Custom Log Name")
+            .description("Log Analytics Custom Log Name")
+            .required(false)
+            .defaultValue("nifimetrics")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+    static final PropertyDescriptor LINUX_PRIMARY_KEY = new PropertyDescriptor.Builder()
+            .name("Linux Primary Key")
+            .description("Connected Sources - Linux Primary Key")
```

Comment:
Can you elaborate a bit on the description? I'm not sure that I understand what this Property is used for.

Reply:
This is the shared key getting from azure log analytic workspace. https://www.systemcenterautomation.com/2018/05/find-azure-log-analytics-keys/

Reply:
Changed to LOG_ANALYTICS_WORKSPACE_KEY